modules/dragster/src/main/ts/ephox/dragster/datatransfer/Event.ts

```diff
@@ -0,0 +1,45 @@
+import { Id, Optional } from '@ephox/katamari';
+
+export const enum Event {
+  Dragstart,
+  Dragend,
+  Drop
+}
+
+const eventId = Id.generate('event');
+
+const getEvent = (transfer: DataTransfer): Optional<Event> => {
+  const dt: any = transfer;
+  return Optional.from(dt[eventId]);
+};
+
+const mkSetEventFn = (type: Event) => (transfer: DataTransfer): void => {
+  const dt: any = transfer;
+  dt[eventId] = type;
+};
+
+const setEvent = (transfer: DataTransfer, type: Event): void => mkSetEventFn(type)(transfer);
+
+const setDragstartEvent = mkSetEventFn(Event.Dragstart);
+const setDropEvent = mkSetEventFn(Event.Drop);
+const setDragendEvent = mkSetEventFn(Event.Dragend);
+
+const checkEvent = (expectedType: Event) => (transfer: DataTransfer): boolean => {
+  const dt: any = transfer;
```

Comment:
Maybe try to avoid `any` you treat this as a record so should be `Record<string, any>` or `Record<string, unknown>`.

Reply:
Thanks for the suggestion, will keep this in mind for next time.